metrics/web/lib/auth/presentation/state/auth_notifier.dart

```diff
@@ -58,20 +75,46 @@ class AuthNotifier extends ChangeNotifier {
 
   /// Signs in user to the app using an [email] and a [password].
   Future<void> signInWithEmailAndPassword(String email, String password) async {
-    _authExceptionDescription = null;
+    _authErrorMessage = null;
     notifyListeners();
 
     try {
       await _signInUseCase(UserCredentialsParam(
-        email: email,
-        password: password,
+        email: Email(email),
+        password: Password(password),
       ));
     } on AuthenticationException catch (exception) {
-      _authExceptionDescription = AuthErrorMessage(exception.code);
+      _authErrorMessage = AuthErrorMessage(exception.code);
       notifyListeners();
     }
   }
 
+  /// Validates the given [value] as an email.
+  void validateEmail(String value) {
+    _emailValidationErrorMessage = null;
+
+    try {
+      Email(value);
+    } on EmailValidationException catch (exception) {
+      _emailValidationErrorMessage = EmailValidationErrorMessage(
+        exception.code,
+      );
+    }
+  }
```

Comment:
What if we try common validator contract - returns String if there is any issue, returns null if everything is ok? Also that means that we can probably create separate utility classes... Let me know what you think 💭